cpp/include/raft/neighbors/detail/cagra/topk_for_cagra/topk_core.cuh

```diff
@@ -302,15 +299,15 @@ __device__ inline void select_best_index_for_next_threshold(
   // index under the condition that the sum of the number of elements found
   // so far ('nx_below_threshold') and the csum value does not exceed the
   // topk value.
-  typedef BlockScan<uint32_t, blockDim_x> BlockScanT;
+  typedef block_scan<uint32_t> BlockScanT;
```

Comment:
I agree with you that it is preferred to rely on cub. Let me check if the register usage of your suggested method is less than or equal to the custom block scan implementation, as it can cause throughput degradation.